### ISSUE TITLE
Setting "fix" to "false" in action configuration is not coerced to a boolean value, so behaves as if the setting was enabled

### ISSUE THREAD

**Issue by thompson-tomo** [DavidAnson/markdownlint-cli2-action#366]
Within my repo I am using the github action to check all pr's which gives us nicely formatted annotations and for local development we use the cli specified via package.json. To validate our local dev environment command we run the local cli command as part of cli when the package.json changes.

I recently discovered that the gh action is not flagging issues when it should've.

The local cli command logs looks like https://github.com/open-telemetry/opentelemetry-ruby-contrib/actions/runs/25949875063/job/76285509575#step:8:1:

```
2026-05-16T02:00:08.6979172Z ##[group]Run npm run check
2026-05-16T02:00:08.8076057Z > check
2026-05-16T02:00:08.8077054Z > npm run check:lint && npm run check:format && npm run check:links && npm run check:spelling && npm run check:source
2026-05-16T02:00:08.9067595Z > check:lint
2026-05-16T02:00:08.9068511Z > npm run check:lint:json && npm run check:lint:markdown && npm run check:lint:ruby
2026-05-16T02:00:09.2350072Z > check:lint:markdown
2026-05-16T02:00:09.2350799Z > markdownlint-cli2 --config .markdownlint-cli2.yaml '**/*.md'
2026-05-16T02:00:09.4471245Z markdownlint-cli2 v0.22.1 (markdownlint v0.40.0)
2026-05-16T02:00:09.4472915Z Finding: **/*.md
2026-05-16T02:00:09.7477903Z Linting: 135 file(s)
2026-05-16T02:00:10.5671785Z Summary: 2 error(s)
2026-05-16T02:00:10.5683728Z instrumentation/trilogy/README.md:52 error MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
2026-05-16T02:00:10.5686052Z instrumentation/trilogy/README.md:53 error MD022/blanks-around-headings Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Above] [Context: "## Compatibility"]
2026-05-16T02:00:10.6176135Z ##[error]Process completed with exit code 1.
```

Which is expected based on my config. However the github action produces https://github.com/open-telemetry/opentelemetry-ruby-contrib/actions/runs/25949875068/job/76285509616#step:3:1:

```
2026-05-16T01:59:20.3034260Z ##[group]Run DavidAnson/markdownlint-cli2-action@ded1f9488f68a970bc66ea5619e13e9b52e601cd
2026-05-16T01:59:20.3036365Z with:
2026-05-16T01:59:20.3037220Z   config: .markdownlint-cli2.yaml
2026-05-16T01:59:20.3038304Z   fix: false
2026-05-16T01:59:20.3039134Z   globs: **/*.md
2026-05-16T01:59:20.3040009Z   separator: 
2026-05-16T01:59:20.3040869Z ##[endgroup]
2026-05-16T01:59:20.4359390Z markdownlint-cli2 v0.22.1 (markdownlint v0.40.0)
2026-05-16T01:59:20.4361523Z Finding: **/*.md
2026-05-16T01:59:20.5437804Z Linting: 135 file(s)
2026-05-16T01:59:21.2704718Z Summary: 0 error(s)
```

Key difference being no errors as can be seen by the logs, the logs also show that the github action config input is the same as the cli.

It looks like the gh action is only enabling some rules even though default is true.  This based on testing in https://github.com/open-telemetry/opentelemetry-ruby-contrib/actions/runs/25951078488/job/76288899706 which has flagged MD041 with MD001 being disabled via Config but hasn't flagged other issues with lists as seen in https://github.com/open-telemetry/opentelemetry-ruby-contrib/actions/runs/25951078552/job/76288899962#step:8:1

The expectation is if the github action and cli use the same config, they produce the same result.

**Comment by DavidAnson** [comment]
I agree with your summary and expectation. Unfortunately, GitHub broke Codespaces a few days ago (perhaps just for Safari?), so I can't easily reproduce this using your repo.

FYI, your configuration is more complicated than it needs to be, though you may have reasons. FYI that `default=true` is unnecessary because all rules are enabled by default. Also, `.markdownlint-cli2.yaml` could include everything from `.markdownlint.yaml` and `.markdownlintignore` within itself and then you'd have just a single configuration file. And because it's using a standard name and in the root of the repo, the `config` parameter should not be needed by either CLI or the Action.

That said, both scenarios should behave the same and I can't tell from brief visual inspection why they're not. Your theory about different rules being in effect is a reasonable one, BUT assuming the configuration is coming from <https://github.com/thompson-tomo/opentelemetry-ruby-contrib/blob/mdlint-fix/.markdownlint.yaml> and the corresponding files in that branch, I don't think the theory is right because none of the configuration I see disables MD031 or MD022 explicitly or implicitly. So both rules should be enabled in both cases whether or not the right configuration is being loaded - thus both scenarios should report the violations on lines 52/53.

I'll experiment a bit now and post an update once I know more.

**Comment by DavidAnson** [comment]
Oooookay, I'm able to reproduce it separately!

https://github.com/DavidAnson/playground/actions/runs/25953045790

**Comment by DavidAnson** [comment]
BINGO! It's your use of `fix: false`. The action is not coercing that setting it receives as the string `"false"` into a boolean and so that is treated as the value `true` and since both violations are fixable, it quietly fixes them and then has nothing left to report. Because the default behavior is to NOT fix, you can remove that setting from your action to work around this issue. And I will fix this bug in the action. :)

Thanks for reporting this!

**Comment by thompson-tomo** [comment]
Thanks @DavidAnson i agree it is a very strange one and thanks for your prompt investigation/guidance.

**Comment by DavidAnson** [comment]
FYI, you can see all the simplification I suggested here (view history to see each applied in turn): https://github.com/DavidAnson/playground/tree/markdownlint-cli2-action-issue-366

**Comment by DavidAnson** [comment]
Okay, in my defense, the code DOES try to coerce the value, but...
```
> Boolean("false")
true
```

So I am not a clown, and I do have a test for "fix", but I only test the "true" case. :|

**Comment by thompson-tomo** [comment]
No worries, I appreciate your prompt activity on this. 🙂

**Comment by DavidAnson** [comment]
FYI, I do not think I will do a new release just for this fix given how unlikely it has been in practice. I would recommend you remove "fix" from your YAML or alternatively, you should be able to use the empty string value to mean false.

**Comment by DavidAnson** [comment]
v24.0.0
